ruma-common/src/push.rs

```diff
@@ -297,6 +393,28 @@ impl From<ConditionalPushRuleInit> for ConditionalPushRule {
     }
 }
 
+impl ConditionalPushRule {
+    /// Check if the push rule applies to the event.
+    ///
+    /// # Arguments
+    ///
+    /// * `event` - A map containing the flattened JSON representation of a room message event.
+    /// * `context` - The context of the room at the time of the event.
+    pub fn applies(
+        &self,
+        event: &BTreeMap<String, String>,
```

Comment:
I don't think we should have `BTreeMap<String, String>` as public API. If you want to avoid flattening events more often than necessary, we should have a wrapper type around the internal flattened representation so it's not exposed as part of the public API, and `flatten_json` should be public so users can flatten the JSON using the same function as we do internally.

Alternatively we could make all the new functions except `get_actions` private.